tools/generate-dom-interfaces.js

```diff
@@ -0,0 +1,264 @@
+  { path: './events/ProgressEvent', type: 'ProgressEventImpl', isDefault: true, name: 'ProgressEvent' },
+  { path: './events/TouchEvent', type: 'TouchEventImpl', isDefault: true, name: 'TouchEvent' },
+  { path: './events/UIEvent', type: 'UIEventImpl', isDefault: false, name: 'UIEvent' },
+  // Others
+  { path: './domexception', type: 'DOMExceptionImpl', isDefault: true, name: 'DOMException' },
+  { path: './custom-elements/CustomElementRegistry', type: 'CustomElementRegistryImpl', isDefault: false, name: 'CustomElementRegistry' },
+  { path: './hr-time/Performance', type: 'PerformanceImpl', isDefault: false, name: 'Performance' },
+  { path: './range/AbstractRange', type: 'AbstractRangeImpl', isDefault: false, name: 'AbstractRange' },
+  { path: './range/Range', type: 'RangeImpl', isDefault: false, name: 'Range' },
+  { path: './mutation-observer/MutationObserver', type: 'MutationObserverImpl', isDefault: false, name: 'MutationObserver' },
+  { path: './mutation-observer/MutationRecord', type: 'MutationRecordImpl', isDefault: false, name: 'MutationRecord' },
+  { path: './crypto/Noise', type: 'NoiseImpl', isDefault: true, name: 'Noise' },
+  { path: './geometry/DOMPoint', type: 'DOMPointImpl', isDefault: true, name: 'DOMPoint' },
+  { path: './geometry/DOMPointReadOnly', type: 'DOMPointReadOnlyImpl', isDefault: true, name: 'DOMPointReadOnly' },
+  { path: './geometry/DOMRect', type: 'DOMRectImpl', isDefault: true, name: 'DOMRect' },
+  { path: './geometry/DOMRectReadOnly', type: 'DOMRectReadOnlyImpl', isDefault: true, name: 'DOMRectReadOnly' },
+  { path: './geometry/DOMMatrix', type: 'DOMMatrixImpl', isDefault: true, name: 'DOMMatrix' },
+  { path: './image/ImageData', type: 'ImageDataImpl', isDefault: true, name: 'ImageData' },
+  // WebXR
+  { path: './xr/XRPose', type: 'XRPoseImpl', isDefault: true, name: 'XRPose' },
+  { path: './xr/XRRigidTransform', type: 'XRRigidTransformImpl', isDefault: true, name: 'XRRigidTransform' },
+  { path: './xr/XRSession', type: 'XRSessionImpl', isDefault: true, name: 'XRSession' }
+];
+
+// Build the templates
+const buildTypeImports = (arg) => {
+  if (arg.IS_DEFAULT) {
+    const baseTemplate = defaultTemplate(`
+      import type ${arg.TYPE} from '${arg.SPECIFIER}';
+    `);
+    return baseTemplate({});
+  } 
+  const baseTemplate = defaultTemplate(`
+    import type { ${arg.TYPE} } from '${arg.SPECIFIER}';
+  `);
+  return baseTemplate({});
+};
+
+const buildHeadStatement = defaultTemplate(`
+  TYPE_IMPORTS
+  let implementationLoaded = false;
+  const setInterfaces = new Map<string, any>();
+`);
+
+/**
+ * solve the issue https://github.com/jestjs/jest/issues/11434 
+ * by importing modules dynamically either in parallel or sequentially based on the isParallel flag.
+ * @param running parallel or not
+ */
+
+const buildParallelImports = (arg) => {
+  const baseTemplate = defaultTemplate(`
+    import('${arg.MODULE}')
+  `);
+  return baseTemplate({});
+}
```

Comment:
```suggestion
};
```

This is a statement, it should be ended with a semi-colon.